packages/compass-settings/src/stores/settings.ts

```diff
@@ -67,13 +112,21 @@ export const reducer: Reducer<State> = (state = INITIAL_STATE, action) => {
       return {
         ...state,
         isModalOpen: true,
+        tab: action.tab ?? state.tab,
+      };
+    case ActionTypes.SelectTab:
+      return {
+        ...state,
+        tab: action.tab ?? state.tab,
       };
     case ActionTypes.CloseSettingsModal:
       return {
         ...state,
         isModalOpen: false,
+        tab: undefined,
       };
     case ActionTypes.ChangeFieldValue:
+      if (state.loadingState !== 'ready') return { ...state };
```

Comment:
Nit: no need for shallow cloning if state is not changed

```suggestion
      if (state.loadingState !== 'ready') return state;
```

Reply:
Done in 8236817e9e55c318f7e578a80722b8310c0f7742 :+1: